src/main/java/tc/oc/pgm/listeners/ModerationListener.java

```diff
@@ -0,0 +1,97 @@
+package tc.oc.pgm.listeners;
+
+import java.util.Date;
+import javax.annotation.Nullable;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.BanList;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import tc.oc.component.Component;
+import tc.oc.component.types.PersonalizedText;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.named.NameStyle;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.commands.ModerationCommands;
+import tc.oc.pgm.commands.ModerationCommands.PunishmentType;
+import tc.oc.pgm.events.PlayerPunishmentEvent;
+import tc.oc.pgm.events.PlayerTimedPunishmentEvent;
+import tc.oc.util.components.Components;
+
+public class ModerationListener implements Listener {
```

Comment:
I am generally not sure if this should be handled in a listener, or if it should just be directly handled in the ModerationCommands, but you have to pick one and commit to it.

 - If you want to kill this class, just have the commands send the event and if not cancelled, apply the punishment & broadcast (All on ModerationCommands).

 - If you want to keep this class, don't perform the player kick on the moderation commands, just launch the event there, and if cancelled return a message to the user. Apply the punishment & broadcast it here, if event isn't cancelled.